### PR TITLE
Add Slasher Simulator to E2E

### PR DESCRIPTION
**Pull request by 0xKiwi**
This PR adds the slasher simulator to E2E testing in order to confirm that the new slasher is capable of detecting attestations/proposals accurately.

### CODE REVIEW

File: cmd/beacon-chain/slasher-simulator/simulator.go [hunk 1]
```diff
@@ -30,6 +32,13 @@ var Commands = &cli.Command{
 	}),
 	Before: func(cliCtx *cli.Context) error {
 		runtime.GOMAXPROCS(runtime.NumCPU())
+
+		logFileName := cliCtx.String(cmd.LogFileName.Name)
```
Comment (rauljordan) [inline cmd/beacon-chain/slasher-simulator/simulator.go:36]: Not a fan of doing stuff in these cli entrypoint files. Can we move this to the actual simulator code?

File: endtoend/components/slasher_simulator.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	"github.com/prysmaticlabs/prysm/endtoend/helpers"
+	e2e "github.com/prysmaticlabs/prysm/endtoend/params"
+)
+
+// StartSlasherSimulator runs a beacon chain client with the slasher simulator enabled to simulate slashing detection.
+func StartSlasherSimulator(t *testing.T) {
+	binaryPath, found := bazel.FindBinary("cmd/beacon-chain", "beacon-chain")
+	if !found {
+		t.Log(binaryPath)
+		t.Fatal("Beacon chain binary not found")
```
Comment (rauljordan) [inline endtoend/components/slasher_simulator.go:19]: Edit

File: endtoend/endtoend_test.go [hunk 3]
```diff
@@ -118,6 +119,19 @@ func runEndToEndTest(t *testing.T, config *e2etypes.E2EConfig) {
 		}
 	}
 
+	simulatorOutFile, err := os.Open(path.Join(e2e.TestParams.LogPath, e2e.SlasherSimulatorLogFileName))
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Run("slasher simulator completed", func(t *testing.T) {
```
Comment (rauljordan) [inline endtoend/endtoend_test.go:126]: We should check that there were no errors as well

File: endtoend/components/slasher_simulator.go (hunk 2 above)
Comment (0xKiwi) [reply endtoend/components/slasher_simulator.go:19]: It's accurate, it uses the beacon chain binary.

File: endtoend/endtoend_test.go [hunk 4]
```diff
@@ -118,6 +119,22 @@ func runEndToEndTest(t *testing.T, config *e2etypes.E2EConfig) {
...
+		if err = helpers.WaitForTextInFile(simulatorOutFile, "Correctly detected simulated attester slashing"); err != nil {
+			t.Fatalf("could not find detection logs for slasher, this means it had issues with detection: %v", err)
+		}
+		if err = helpers.WaitForTextInFile(simulatorOutFile, "Correctly detected simulated proposer slashing"); err != nil {
+			t.Fatalf("could not find detection logs for slasher, this means it had issues with detection: %v", err)
+		}
+		if err = helpers.CheckTextNotInFile(simulatorOutFile, "Did not detect"); err != nil {
```
Comment (rauljordan) [inline endtoend/endtoend_test.go:133]: We can also check for the word ERROR right?

**Review by rauljordan — APPROVED** [review]

File: endtoend/endtoend_test.go (hunk 4 above)
Comment (0xKiwi) [reply endtoend/endtoend_test.go:133]: Ah I'll add the errors here to the finishing logs, I'll leave the checks as is though

**Review by rauljordan — APPROVED** [review]

**Pull request merged**